docs/developers/sc-calls.md

```diff
@@ -0,0 +1,297 @@
+
+A `sync` call executes immediately and is used when the source and destination contracts are `on the same shard`. This is the fastest type of call and is ideal for high-speed operations. Because of that, it's recommended that multiple contracts that are part of the same mechanism should lie on the same shard. For example, a sync call is commonly used to deploy a smart contract.
+
+An `async` call is used when the source and destination contracts may reside `on different shards`. When in doubt about the destination shard, async calls are recommended to maximize transaction success. An example of a popular async call is the upgrade of a smart contract.
+
+:::note
+Async call functions do not return values (they return the `never` type or nothing) and may include a `callback` function to handle the response from the destination contract.
+:::
+
+This is an example of a callback function that gets triggered after an `issue_fungible` action:
+
+```rust
+    #[callback]
+    fn esdt_issue_callback(
+        &self,
+        caller: &ManagedAddress,
+        #[call_result] result: ManagedAsyncCallResult<()>,
+    ) {
+        let (token_identifier, returned_tokens) = self.call_value().egld_or_single_fungible_esdt();
+        // callback is called with ESDTTransfer of the newly issued token, with the amount requested,
+        // so we can get the token identifier and amount from the call data
+        match result {
+            ManagedAsyncCallResult::Ok(()) => {
+                self.last_issued_token().set(token_identifier.unwrap_esdt());
+                self.last_error_message().clear();
+            },
+            ManagedAsyncCallResult::Err(message) => {
+                // return issue cost to the caller
+                if token_identifier.is_egld() && returned_tokens > 0 {
+                    self.tx().to(caller).egld(&returned_tokens).transfer();
+                }
+
+                self.last_error_message().set(&message.err_msg);
+            },
+        }
+    }
+```
+
+Each of these calls further divide in multiple cathegories, depending on the mechanism they use, different developer use-cases and expected results.
+
+[comment]: # (mx-context-auto)
+
+## Sync calls
+
+Sync calls are used when `intra-shard communication` is required, allowing transactions to execute in the same shard only.
+There are four main types of sync calls developers can use inside a smart contract.
+
+[comment]: # (mx-context-auto)
+
+### Sync call
+
+A standard sync call performs a direct, synchronous transaction to a contract on the same shard. This type of call is launched with `.sync_call()`.
+
+```rust
+    /// Executes transaction synchronously.
+    ///
+    /// Only works with contracts from the same shard.
+    pub fn sync_call(self) -> <RH::ListReturns as NestedTupleFlatten>::Unpacked
+```
+
```

Comment:
Would be helpful to say that this is an example and then explain the transaction